### PR TITLE
[WIP] Parser for Dhall Expressions (#4)

### PR DESCRIPTION
**Pull request by missingfaktor**
So @amarpotghan and @tejinders, three of us – @chandan911, @rhushikesh, and I – had a meeting today. We had a brief project walkthrough, and then started working on the parser. What you see in the PR is as far as we got. Which is not a lot, but it is a start! @rhushikesh and I will continue working on parsing. @chandan911 has picked up a couple other stories.

### CODE REVIEW

File: src/main/scala/dhall/Path.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package dhall
+
+trait Path {
+  val value: String
```
Comment (missingfaktor) [inline src/main/scala/dhall/Path.scala]: Make this a `def`

File: src/main/scala/dhall/DhallParser.scala [hunk 2]
```diff
@@ -12,7 +12,7 @@ class DhallParser private[dhall](val input: ParserInput) extends Parser {
   }
 
   def Expression: Rule1[Embed[Path]] = rule {
-    EnvExpression | UrlExpression
+    EnvExpression | UrlExpression | FIleExpression
```
Comment (missingfaktor) [inline src/main/scala/dhall/DhallParser.scala]: Typo

File: src/main/scala/dhall/DhallParser.scala [hunk 3]
```diff
@@ -7,16 +7,16 @@ import scala.util.Try
```
Comment (missingfaktor) [inline src/main/scala/dhall/DhallParser.scala]: In the commit message header, use `Add`. Not `Added`. Imperative mood.

File: src/main/scala/dhall/Path.scala (hunk 1 above)
Comment (rhushikesh) [reply src/main/scala/dhall/Path.scala]: yes

File: src/main/scala/dhall/DhallParser.scala (hunk 2 above)
Comment (rhushikesh) [reply src/main/scala/dhall/DhallParser.scala]: done

**Comment by missingfaktor** [comment]
@rhushikesh Use imperative mood in commit message headers. `Added` -> `Add`, `Corrected` -> `Correct`, `Correcting` -> `Correct` etc. The rationale is that grammatical forms such as gerunds, past tense, and so on eat up characters in an already limited space (72 characters), and bring no additional clarity.

**Pull request merged**